llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp

```diff
@@ -2320,7 +2326,7 @@ bool RISCVAsmParser::generateVTypeError(SMLoc ErrorLoc) {
   return Error(
       ErrorLoc,
       "operand must be "
-      "e[8|16|32|64],m[1|2|4|8|f2|f4|f8],[ta|tu],[ma|mu]");
+      "e{8|16|32|64},[m{1|2|4|8|f2|f4|f8},]{ta|tu},{ma|mu}");
```

Comment:
Why change `[` to `{`? Mask/tail policies are optional I think. But the original format is not right, it should be `e{8|16|32|64}[,m{1|2|4|8|f2|f4|f8}][,{ta|tu}][,{ma|mu}]` now?

Reply:
1. https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp#L1651 -- here `[` means optional so you're right that original format is not right. This is why I've changed `[` to `{`.
2. As far as I can see from RISC-V spec mask/tail policies are not optional. GCC also doesn't think so: https://godbolt.org/z/54h1Kf3a1

Reply:
Actually GCC(binutils) supports optional mask/tail policies: https://godbolt.org/z/GPEG8xzrf
It's a known issue for LLVM if I remember correctly, we may have some discussion before but I can't find the link.
Binutils supports such thing just for compatibility with old version (0.7.1, pity) of RVV.

Reply:
IMHO, it's better to stick with the current changes since looks like RVV 1.0 doesn't support optional mask/tail policies, and someone who is interested in compatibility with the RVV 0.7.1 can do changes above this PR.
Anyway, I would like to hear your opinion on this.

Reply:
> All right, please use e{8|16|32|64}[,m{1|2|4|8|f2|f4|f8}],{ta|tu},{ma|mu} (put [ in front of ,) in case we may support optional policies someday.

Reply:
Addressed.